---
title: "Simran"
author: "Simran"
date: "05/03/2020"
output: md_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r packages}
library(magrittr)
library(dplyr)
library(tidyverse)
library(ggplot2)
library(reshape2)
library(edgeR)
library(pheatmap)
library(ggplot2)
```

```{r read in data} 
demo<-read.csv("/Users/simransamra/R/git_temp/Repo_team_Genome-Surfers_W2020/data/raw_data/tcga_paad_clinical.csv", header = T)
load("/Users/simransamra/R/git_temp/Repo_team_Genome-Surfers_W2020/data/raw_data/tcga_paad.RData")

#Melt data
tcga$gene <- rownames(tcga)
dataMelt<-melt(tcga, id.vars = "gene", var = "Samples")

#Set up data
demo <- demo %>% select(c(submitter_id, age_at_index,year_of_birth, year_of_death, vital_status, race, gender, ajcc_pathologic_m, ajcc_pathologic_n,ajcc_pathologic_t, ajcc_pathologic_stage))

tcgaT <- as.data.frame(t(as.matrix(tcga)))
tcgaN <- tibble::rownames_to_column(tcgaT, "submitter_id")

demo$submitter_id <- as.factor(demo$submitter_id)
data <- right_join(x = tcgaN, y = demo, by = "submitter_id")

```

```{r fig1}
ggplot(data, aes(x = age_at_index)) + 
geom_density(aes(fill = vital_status), alpha = 0.5) +
xlab("Age at Index")  + 
ylab("Density") +
labs(fill='Vital Status') +
scale_fill_manual(values = c("darkblue", "darkred"))
```

```{r fig2}
ggplot(data, aes(x = age_at_index)) + 
geom_density(aes(fill = ajcc_pathologic_m), alpha = 0.5) +
xlab("Age at Index")  + 
ylab("Density") +
labs(fill='AJCC Pathologic M') 
```


```{r fig3, fig.width = 10, fig.asp = .85}
#Boxplot - Distribution of Gene Expression
ggplot(dataMelt, aes(x=Samples, y=value)) + 
   geom_boxplot() + 
   xlab("Samples")  + 
  ylab("Expression (Log2 Transformed)")+ 
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  ggtitle("Distribution of Gene Expression")+
  theme(plot.title = element_text(hjust = 0.5))
```

```{r fig4, fig.width = 10, fig.asp = .90}
#Density Plot
ggplot(dataMelt, aes(value, color = Samples)) + 
  geom_density() + 
  xlab("Expression (Log2 Transformed)") +
  ylab("Density")+ 
  ggtitle("Distribution of Gene Expression") +
  theme(plot.title = element_text(hjust = 0.5)) + theme(legend.position="none")
```


```{r fig5, fig.asp = .95}
#Prepare Data
load("/Users/simransamra/R/git_temp/Repo_team_Genome-Surfers_W2020/data/raw_data/tcga_paad.RData")
#tcga$gene <- rownames(tcga)
demoHeat<-demo
toDelete <- seq(1, nrow(demoHeat), 2)
demoHeat<-demoHeat[toDelete ,]

designFactors <- as.data.frame(demoHeat[, c("ajcc_pathologic_m",  "ajcc_pathologic_n", "ajcc_pathologic_t", "ajcc_pathologic_stage")])

rownames(designFactors) <- colnames(tcga)
data.matrix <- cor(tcga)
pheatmap(data.matrix, cluster_rows = T, scale = "none", clustering_method = "average", 
    clustering_distance_cols = "correlation", show_colnames = T, show_rownames = T, 
    main = "Clustering heatmap", annotation = designFactors, treeheight_col = 35, treeheight_row = 35,
    fontsize = 10)
```

```{r fig6, fig.asp = .95}
#Prepare Data
load("/Users/simransamra/R/git_temp/Repo_team_Genome-Surfers_W2020/data/raw_data/tcga_paad.RData")
#tcga$gene <- rownames(tcga)
demoHeat<-demo
toDelete <- seq(1, nrow(demoHeat), 2)
demoHeat<-demoHeat[toDelete ,]

designFactors <- as.data.frame(demoHeat[, c("race",  "gender")])

rownames(designFactors) <- colnames(tcga)
data.matrix <- cor(tcga)
pheatmap(data.matrix, cluster_rows = T, scale = "none", clustering_method = "average", 
    clustering_distance_cols = "correlation", show_colnames = T, show_rownames = T, 
    main = "Clustering heatmap", annotation = designFactors, treeheight_col = 35, treeheight_row = 35,
    fontsize = 10)
```

```{r fig7, fig.asp = .95}
#Prepare Data
load("/Users/simransamra/R/git_temp/Repo_team_Genome-Surfers_W2020/data/raw_data/tcga_paad.RData")
#tcga$gene <- rownames(tcga)
demoHeat<-demo
toDelete <- seq(1, nrow(demoHeat), 2)
demoHeat<-demoHeat[toDelete ,]

designFactors <- as.data.frame(demoHeat[, c("vital_status")])

rownames(designFactors) <- colnames(tcga)
data.matrix <- cor(tcga)
pheatmap(data.matrix, cluster_rows = T, scale = "none", clustering_method = "average", 
    clustering_distance_cols = "correlation", show_colnames = T, show_rownames = T, 
    main = "Clustering heatmap", annotation = designFactors, treeheight_col = 35, treeheight_row = 35,
    fontsize = 10)
```